src/error.rs

```diff
@@ -126,6 +133,13 @@ impl fmt::Display for Error {
             &Error::ResponseError(ref inner) => inner.fmt(fmt),
             &Error::CursorNotFoundError => write!(fmt, "No cursor found for cursor operation."),
             &Error::PoisonLockError => write!(fmt, "Socket lock poisoned while attempting to access."),
+            &Error::CodedError(ref err) => write!(fmt, "{}", err),
+            &Error::EventListenerError(ref err) => match err {
+                &Some(ref e) => write!(fmt, "Due to a poisoned lock on the \
```

Comment:
nittiest of the nitpicks: "Unable to emit failure due to poisoned lock" is more concise. Commas in error messages read poorly.